sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSinkLog.scala

```diff
@@ -45,7 +46,20 @@ case class SinkFileStatus(
     modificationTime: Long,
     blockReplication: Int,
     blockSize: Long,
-    action: String) {
+    action: String,
+    commitTime: Long) {
+
+  def this(
+      path: String,
+      size: Long,
+      isDir: Boolean,
+      modificationTime: Long,
+      blockReplication: Int,
+      blockSize: Long,
+      action: String) {
+    // use modification time if we don't know about exact commit time
```

Comment:
If we concern about Hadoop timestamp / timezone, we can just set it to `Long.MaxValue`. This means existing entries in metadata will not be affected by the change.

Reply:
Just changed it. Retention logic completely relies on commit timestamp and not relevant to individual file's modification time.